Specs/Scene/TileBoundingS2CellSpec.js

```diff
@@ -0,0 +1,185 @@
+      token: "1",
+    });
+    expect(tS2Cell).toBeDefined();
+    expect(tS2Cell.boundingVolume).toBeDefined();
+    expect(tS2Cell.boundingSphere).toBeDefined();
+    expect(tS2Cell.s2Cell).toBeDefined();
+    expect(tS2Cell.center).toBeDefined();
+    expect(tS2Cell.minimumHeight).toBeDefined();
+    expect(tS2Cell.maximumHeight).toBeDefined();
+  });
+
+  it("can be instantiated with S2 cell and heights", function () {
+    var tS2Cell = new TileBoundingS2Cell(s2Options);
+    expect(tS2Cell).toBeDefined();
+    expect(tS2Cell.boundingVolume).toBeDefined();
+    expect(tS2Cell.boundingSphere).toBeDefined();
+    expect(tS2Cell.s2Cell).toBeDefined();
+    expect(tS2Cell.center).toBeDefined();
+    expect(tS2Cell.minimumHeight).toBeDefined();
+    expect(tS2Cell.maximumHeight).toBeDefined();
+  });
+
+  it("distanceToCamera throws when frameState is undefined", function () {
+    expect(function () {
+      return tileS2Cell.distanceToCamera();
+    }).toThrowDeveloperError();
+  });
+
+  it("distance to camera is 0 when camera is inside bounding volume", function () {
+    camera.position = s2Cell.getCenter();
+    expect(tileS2Cell.distanceToCamera(frameState)).toEqual(0.0);
+  });
+
+  it("distance to camera for top and bottom planes", function () {
+    var testDistance = 1;
+    var plane = Plane.clone(tileS2Cell._boundingPlanes[0]);
+    plane.distance -= testDistance;
+    camera.position = Plane.projectPointOntoPlane(plane, tileS2Cell.center);
+    expect(tileS2Cell.distanceToCamera(frameState)).toEqualEpsilon(
+      testDistance,
+      CesiumMath.EPSILON6
+    );
+
+    plane = Plane.clone(tileS2Cell._boundingPlanes[1]);
+    plane.distance -= testDistance;
+    camera.position = Plane.projectPointOntoPlane(plane, tileS2Cell.center);
+    expect(tileS2Cell.distanceToCamera(frameState)).toEqualEpsilon(
+      testDistance,
+      CesiumMath.EPSILON6
+    );
+  });
+
+  var edgeOneScratch = new Cartesian3();
+  var edgeTwoScratch = new Cartesian3();
+  var planeCenterScratch = new Cartesian3();
+  it("distance to camera for side planes", function () {
+    var testDistance = 100000;
+    for (var i = 0; i < 4; i++) {
+      var plane = Plane.clone(tileS2Cell._boundingPlanes[2 + i]);
+
+      var edgeOne = Cartesian3.midpoint(
+        tileS2Cell._vertices[i % 4],
+        tileS2Cell._vertices[4 + i],
+        edgeOneScratch
+      );
+
+      var edgeTwo = Cartesian3.midpoint(
+        tileS2Cell._vertices[4 + ((i + 1) % 4)],
+        tileS2Cell._vertices[(i + 1) % 4],
+        edgeTwoScratch
+      );
+
+      var planeCenter = Cartesian3.midpoint(
+        edgeOne,
+        edgeTwo,
+        planeCenterScratch
+      );
+
+      plane.distance -= testDistance;
+      camera.position = Plane.projectPointOntoPlane(plane, planeCenter);
+      expect(tileS2Cell.distanceToCamera(frameState)).toEqualEpsilon(
+        testDistance,
+        CesiumMath.EPSILON6
+      );
+    }
+  });
+
+  it("distanceToCamera", function () {
```

Comment:
I get it, hard to set up these test cases given the weird geometry, but some suggestions of things you could add to improve the variety of tests:

1. distance to camera for point on side edge, not just top
2. use bottom vertex instead of top for the second test
3. Also try a point that's not exactly on a vertex but in that "outside the corner" region around it

Reply:
I added test cases for each scenario.